docs/Support/faq.md

```diff
@@ -66,31 +146,109 @@ Yes! See [boolean logic](/accessControl/conditionTypes/booleanLogic) for example
+
+### 9. How to pass the result of LitJsSdk.encryptString(str) as a string param of a Solidity function?
+
+Since the result is a Blob, you have to convert it to a base64 string. Use the code snippet for that:
+
+```js
+// Convert blob to base64 to pass as a string to Solidity
+const blobToBase64 = (blob) => {
+ const reader = new FileReader();
+ reader.readAsDataURL(blob);
+ return new Promise((resolve) => {
+   reader.onloadend = () => {
+     resolve(reader.result);
+   };
+ });
+};
+const encryptedStringBase64 = await blobToBase64(encryptedStringBlob);
+```
+
+### 10. How to pass a string stored in a contract to LitJsSdk.decryptString()?
+
+Since the string is stored as a base64 string you have to convert it to a Blob to be passed to the `decryptString` function. Use the code snippet below for that:
+
+```js
+// Convert base64 to blob to pass in the litSDK decrypt function
+const encryptedStringBlob = await (await fetch(encryptedStringBase64)).blob();
+```
+
+### 11. What’s the maximum number of accessControlConditions allowed at once?
+
+30
+
+### 12. Where can I save the encryptedString & encryptedSymmetricKey?
+
+The Lit network doesn’t store these encrypted contents for you. You can store these anywhere you want; in a database, in an on-chain smart contract, IPFS, anywhere else you like.
+
+### 13. How to construct an accessControlCondition to authorize only a specific wallet address?
+
+Check out the solution our docs [here](https://developer.litprotocol.com/coreConcepts/accessControl/EVM/basicExamples#a-specific-wallet-address).
+
+### 14. How to use a time-lock based accessControlCondition?
+
+Check out the solution our docs [here](https://developer.litprotocol.com/coreConcepts/accessControl/EVM/timelock).
+
+<br />
+
+## Design Patterns
+
+### 1. How to allow only permitted users to execute a LitAction?
+
+Use the `PKPPermission.addPermittedAddress()` to allow only specific users to execute a LitAction. Note, this will allow these users to execute any LitAction.
+
+For other permissions, please see the contract [here](https://github.com/LIT-Protocol/LitNodeContracts/blob/main/contracts/PKPPermissions.sol).
+
+### 2. How to allow permitted users to execute only specific LitActions?
+
+Assign the PKP to itself as we don’t want the PKP owner to arbitrarily change the functioning of the LitAction. Use the PKPHelper.`mintNextAndAddAuthMethods()` for that & to initially setup allow only specific IPFS CIDs to execute.
+
+But anyone can now call these LitActions, so how should one add a permitted list of users? We can store the permitted list of users, either on-chain or the LitAction, can fetch it from there or put that in the LitAction itself and use `conditionalSigning` to check whether the provided AuthSig is permitted to execute the LitAction. Learn more [here](https://developer.litprotocol.com/coreconcepts/litactionsandpkps/actions/conditionalsigning/).
```

Comment:
broken link... not sure we have an example for this other than the conditional signing replit repo (which I don't think explains this exactly)

Reply:
ty @debbly - updated!